src/core_modules/capture-core/components/Pages/MainPage/WorkingLists/WorkingListsContextBuilder.component.js

```diff
@@ -176,20 +210,24 @@ const WorkingListsContextBuilder = (props: Props) => {
         <ManagerContext.Provider
             value={managerData}
         >
-            <EventListLoaderContext.Provider
-                value={eventListData}
+            <ListViewConfigContext.Provider
+                value={listViewConfigContextData}
             >
-                <EventListConfigContext.Provider
-                    value={eventListConfig}
+                <ListViewLoaderContext.Provider
+                    value={listViewLoaderContextData}
                 >
-                    <TemplatesLoader
-                        {...passOnProps}
-                        templates={templates}
-                        listId={listId}
-                        dirtyTemplates={!!dirtyTemplatesStateFirstRunRef.current}
-                    />
-                </EventListConfigContext.Provider>
-            </EventListLoaderContext.Provider>
+                    <ListViewBuilderContext.Provider
+                        value={listViewBuilderContextData}
+                    >
+                        <TemplatesLoader
+                            {...passOnProps}
+                            templates={templates}
+                            listId={listId}
+                            dirtyTemplates={!!dirtyTemplatesStateFirstRunRef.current}
+                        />
+                    </ListViewBuilderContext.Provider>
+                </ListViewLoaderContext.Provider>
+            </ListViewConfigContext.Provider>
```

Comment:
What are your motivations for keeping the file as it is now? 

My point is that the file will look much more approachable and easy to understand if there are only providers. Again this way we can "hide" the implementation details from the reader and let them dig into them only if they wanted.

Reply:
I don't think I get what you mean here. Where does the listViewConfigContextData get it's data from in your example above? is it `export const ListViewConfigContextProvider = (**props: Props**) => {`? What is props? Everything that is passed into the ContextBuilder? In that case it would be hard to know what should be passed on as passOnProps to the TemplatesLoader and complicate things. Could use render props, but not worth the effort I feel.

Reply:
My apologies. I feel that I didnt explain myself very well. Will to give a better example this time. 

```
import React, { useState, createContext } from "react";

// Create Context Object
export const ListViewConfigContext = createContext();

// Create a provider for components to consume and subscribe to changes
export const ListViewConfigContextProvider = ({
        listMeta,
        columnOrder,
        onAddTemplate,
        onUpdateTemplate,
        onDeleteTemplate,
        convertToEventFilterQueryCriteria,
    }) => {
      const listViewConfigContextData = React.useMemo(() => ({
        listMeta,
        columnOrder,
        onAddTemplate,
        onUpdateTemplate,
        onDeleteTemplate,
        convertToEventFilterQueryCriteria,
    }), [
        listMeta,
        columnOrder,
        onAddTemplate,
        onUpdateTemplate,
        onDeleteTemplate,
        convertToEventFilterQueryCriteria,
    ]);

  return (
    <ListViewConfigContext.Provider value={listViewConfigContextData}>
      {props.children}
    </ListViewConfigContext.Provider>
  );
};
```
This way you can have these "provider" components into doing their own thing, which is creating context providers.
They do that in isolation and we keep this concern on that file only.

Is this answering your question?

Reply:
Got it! Thanks. Made an implementation that is very close to your suggestion (made a new PR) https://github.com/dhis2/capture-app/pull/1151